### PR TITLE
Upgrade to LLVM main

### PR DESCRIPTION
**Pull request by ptersilie**
Upgrading to LLVM main broke various things. 53d37db89daa9be555e91052b1405952832b24b8 deals with changes to the basic block map while 049c59205784e199dfc032bf9119b128760d9bcf fixes changes related to opaque pointer mode and the IR.

### CODE REVIEW

**Comment by ltratt** [comment]
Typo in commit https://github.com/ykjit/yk/pull/558/commits/049c59205784e199dfc032bf9119b128760d9bcf s/not longer valid/no longer valid. Please force push an update.

**Comment by ptersilie** [comment]
Pushed.

**Comment by ptersilie** [comment]
Whoops, we have to merge in the ykllvm changes first. Forgot about that. Making a PR now.

File: yksgi/src/lib.rs [hunk 1]
```diff
@@ -13,9 +13,10 @@ use std::slice;
 mod llvmbridge;
 use llvmbridge::{get_aot_original, BasicBlock, Module, Type, Value};
 
-static ALWAYS_SKIP_FUNCS: [&str; 5] = [
+static ALWAYS_SKIP_FUNCS: [&str; 6] = [
     "llvm.lifetime",
     "llvm.lifetime.end.p0i8",
+    "llvm.lifetime.end.p0",
```
Comment (vext01) [inline yksgi/src/lib.rs:19]: I'm uncertain if this is the correct way to do this. I suspect the `.p0` and `.p0i8` suffixes are arbitrary and are part of a mangling scheme.

Should we have another `ALWAYS_SKIP_FUNCS_WITH_PREFIX` list which is contains (for now) the prefix `llvm.lifetime` ?

Comment (vext01) [reply yksgi/src/lib.rs:19]: Since this is in `yksgi`, that is going to be removed (right?) maybe we can ignore this.

Comment (ptersilie) [reply yksgi/src/lib.rs:19]: This is all gonna go away when we switch to stack recovery, so I'm not too fussed that this isn't the best way to handle these cases.

Comment (ltratt) [reply yksgi/src/lib.rs:19]: I tend to agree.

**Comment by vext01** [comment]
Is this ready to go now?

**Comment by ptersilie** [comment]
Yes, let's bors this.

**Comment by vext01** [comment]
The ykllvm default branch was wrong.

@ptersilie updated it.

bors r+

**Comment by vext01** [comment]
The "noir" test failed. Time to put on your detective's hat?

**Comment by vext01** [comment]
Since lukas is busy, I'll try to fix this.

**Comment by vext01** [comment]
I suspect lukas forgot to `git add` this file. There is a small discrepancy between the IR at -O0 and other levels.

@ltratt OK to squash?

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat

**Comment by ptersilie** [comment]
> I suspect lukas forgot to `git add` this file. There is a small discrepancy between the IR at -O0 and other levels.

Sorry. I did forget that indeed. Hope it didn't take too long to find.

**Pull request merged**